files/public.go

```diff
@@ -104,12 +125,19 @@ func (storage *PublishedStorage) LinkFromPool(publishedDirectory string, sourceP
 			return err
 		}
 
-		srcSys := srcStat.Sys().(*syscall.Stat_t)
-		dstSys := dstStat.Sys().(*syscall.Stat_t)
-
-		// source and destination inodes match, no need to link
-		if srcSys.Ino == dstSys.Ino {
-			return nil
+		if storage.linkMethod == LinkMethodCopy {
+			// if source and destination have the same size, no need to copy
+			if srcStat.Size() == dstStat.Size() {
+				return nil
+			}
+		} else {
+			srcSys := srcStat.Sys().(*syscall.Stat_t)
+			dstSys := dstStat.Sys().(*syscall.Stat_t)
+
+			// if source and destination inodes match, no need to link
+			if srcSys.Ino == dstSys.Ino {
```

Comment:
Right, symlink is followed, but if it's symlink  it might point to file on _different_ filesystem with _same_ inode. So that check wouldn't make sense if filesystem for source/destination aren't same.